packages/loaders/src/tiff/index.ts

```diff
@@ -96,59 +97,102 @@ export async function loadOmeTiff(
  *  [{ c: 0, t: 0, z: 0 }, 'https://example.com/channel_0.tif'],
  *  [{ c: 1, t: 0, z: 0 }, 'https://example.com/channel_1.tif'],
- *  [{ c: 2, t: 0, z: 0 }, 'https://example.com/channel_1.tif'],
+ *  [[[0, { c: 2, t: 0, z: 0 }], [1, { c: 3, t: 0, z: 0 }]], 'https://example.com/channels_2-3.tif'],
  * ]);
  *
  * await data.getRaster({ selection: { c: 0, t: 0, z: 0 } });
  * // { data: Uint16Array[...], width: 500, height: 500 }
  *
- * @param {Array<[OmeTiffSelection, (string | File)]>} sources pairs of `[Selection, string | File]` entries indicating the multidimensional selection in the virtual stack in image source (url string, or `File`).
+ * @param {Array<[OmeTiffSelection | [number, OmeTiffSelection][], (string | File)]>} sources
+ * Pairs of `[Selection | [number, Selection][], string | File]` entries indicating the multidimensional selection in the virtual stack in image source (url string, or `File`).
+ * You should only provide [number, Selection][] when loading from stacked tiffs. In this case the number corresponds to the image index in the stack, and the selection is the
+ * selection that image corresponds to.
  * @param {Object} opts
  * @param {GeoTIFF.Pool} [opts.pool] - A geotiff.js [Pool](https://geotiffjs.github.io/geotiff.js/module-pool-Pool.html) for decoding image chunks.
  * @param {string} [opts.name='MultiTiff'] - a name for the "virtual" image stack.
  * @param {Headers=} opts.headers - Headers passed to each underlying fetch request.
  * @return {Promise<{ data: TiffPixelSource[], metadata: ImageMeta }>} data source and associated metadata.
  */
 export async function loadMultiTiff(
-  sources: [OmeTiffSelection, string | (File & { path: string })][],
+  sources: [
+    OmeTiffSelection | [number, OmeTiffSelection][],
+    string | (File & { path: string })
+  ][],
   opts: MultiTiffOptions = {}
 ) {
   const { pool, headers = {}, name = 'MultiTiff' } = opts;
   const tiffImage: MultiTiffImage[] = [];
+  const channelNames = [];
 
   for (const source of sources) {
-    const [selection, file] = source;
+    const [s, file] = source;
+    const imageSelections: [number, OmeTiffSelection][] = Array.isArray(s)
+      ? s
+      : [[0, s]];
     if (typeof file === 'string') {
+      // If the file is a string then we're dealing with loading from a URL.
       const parsedFilename = parseFilename(file);
       const extension = parsedFilename.extension?.toLowerCase();
       if (extension === 'tif' || extension === 'tiff') {
         const tiffImageName = parsedFilename.name;
         if (tiffImageName) {
-          const tiff = await (
-            await fromUrl(file, { headers, cacheSize: Infinity })
-          ).getImage(0);
-          tiffImage.push({ name: tiffImageName, selection, tiff });
+          for (const imageSelection of imageSelections) {
+            const [imageNumber, selection] = imageSelection;
+            const tiff = await (
+              await fromUrl(file, { headers, cacheSize: Infinity })
```

Comment:
why not just open this once and reuse the return of `fromUrl` repeatedly?

Reply:
Good catch, will do!